crates/trie/sparse/src/trie.rs

```diff
@@ -623,6 +623,7 @@ impl<P> RevealedSparseTrie<P> {
                         ));
                         continue
                     }
+                    let retain_updates = self.updates.is_some() && prefix_set_contains(&path);
```

Comment:
`prefix_set_contains` might be called twice (see above). let's assign to var

Reply:
the value is already memoized, so we actually should call `prefix_set_contains` twice